hudi-common/src/main/java/org/apache/hudi/common/model/EmptyHoodieRecordPayload.java

```diff
@@ -18,35 +18,68 @@
 
 package org.apache.hudi.common.model;
 
+import org.apache.hudi.avro.HoodieAvroUtils;
+import org.apache.hudi.common.util.ConfigUtils;
 import org.apache.hudi.common.util.Option;
+import org.apache.hudi.keygen.constant.KeyGeneratorOptions;
 
 import org.apache.avro.Schema;
 import org.apache.avro.generic.GenericRecord;
 import org.apache.avro.generic.IndexedRecord;
 
+import java.io.IOException;
+import java.util.Properties;
+
 /**
  * Empty payload used for deletions.
  */
 public class EmptyHoodieRecordPayload implements HoodieRecordPayload<EmptyHoodieRecordPayload> {
+  private Comparable orderingVal;
 
   public EmptyHoodieRecordPayload() {
+    orderingVal = 0L;
   }
 
-  public EmptyHoodieRecordPayload(GenericRecord record, Comparable orderingVal) {
+  public EmptyHoodieRecordPayload(GenericRecord record, Comparable<?> orderingVal) {
```

Comment:
See my comment in https://github.com/apache/hudi/pull/10277#discussion_r1430513503.  I think we should keep this unchanged, and instead use the actual payload wrapping a null record and ordering value.